exchanges/execute/src/execute.ts

```diff
@@ -0,0 +1,67 @@
+import { pipe, share, filter, map, fromPromise, mergeMap, merge } from 'wonka';
+import {
+  GraphQLSchema,
+  GraphQLFieldResolver,
+  GraphQLTypeResolver,
+  execute,
+} from 'graphql';
+import { Exchange, makeResult, makeErrorResult } from '@urql/core';
+
+interface ExecuteExchangeArgs {
+  schema: GraphQLSchema;
+  rootValue?: any;
+  contextValue?: any;
+  fieldResolver?: GraphQLFieldResolver<any, any>;
+  typeResolver?: GraphQLTypeResolver<any, any>;
+}
+
+const SUPPORTED_OPERATION_TYPES = ['query', 'mutation'];
+
+/** Exchange for executing queries locally on a schema using graphql-js. */
+export const executeExchange = ({
+  schema,
+  rootValue,
+  contextValue,
+  fieldResolver,
+  typeResolver,
+}: ExecuteExchangeArgs): Exchange => ({ forward }) => {
+  return ops$ => {
+    const sharedOps$ = share(ops$);
+
+    const executedOps$ = pipe(
+      sharedOps$,
+      filter(operation =>
+        SUPPORTED_OPERATION_TYPES.includes(operation.operationName)
+      ),
+      map(async operation => {
+        try {
+          const result = await execute(
+            schema,
+            operation.query,
+            rootValue,
+            typeof contextValue === 'function' ? contextValue() : contextValue,
+            operation.variables,
+            operation.operationName,
```

Comment:
I believe this needs to be filled in by this function, which we can just copy over: https://github.com/FormidableLabs/urql/blob/bde241f911a0109abd86b3ecb9e717dfb8243890/packages/core/src/internal/fetchOptions.ts#L13

Due to historical reasons `operationName` is used for two things in `GraphQL`: The operation type and the operation name;

For us the type is `query | subscription | mutation | teardown`, but for GraphQL it sometimes refers to the name that a user has given a query document, e.g. `query MyName`

Reply:
Have wrapped with `getOperationName()`